### PR TITLE
testutil/beaconmock: add head producer

### PR DESCRIPTION
**Pull request by dB2510**
Adds head producer to beaconmock.

category: feature
ticket: #1307

### CODE REVIEW

File: testutil/beaconmock/blockproducer.go [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+const streamID = "block_events"
+
+func NewBlockProducer() *BlockProducer {
+	server := sse.New()
+	server.CreateStream(streamID)
+
+	return &BlockProducer{server: server}
+}
+
+// BlockProducer is stateful structure for providing deterministic block roots based on slot events.
```
Comment (xenowits) [inline testutil/beaconmock/blockproducer.go:42]:
```suggestion
// BlockProducer is a stateful struct for providing deterministic block roots based on slot events.
```

File: testutil/beaconmock/options.go [hunk 2]
```diff
@@ -386,6 +386,12 @@ func WithSyncCommitteeDuties() Option {
 
 			return resp, nil
 		}
+
+		mock.overrides = append(mock.overrides, staticOverride{
+			Endpoint: "/eth/v1/config/spec",
+			Key:      "EPOCHS_PER_SYNC_COMMITTEE_PERIOD",
+			Value:    "2",
```
Comment (corverroos) [inline testutil/beaconmock/options.go:393]: shouldn't this be 1?

File: testutil/validatormock/synccomm.go [hunk 3]
```diff
@@ -225,5 +227,12 @@ func submitSyncMessage(ctx context.Context, eth2Cl eth2wrap.Client, slot eth2p0.
 		})
 	}
 
-	return eth2Cl.SubmitSyncCommitteeMessages(ctx, msgs)
+	err = eth2Cl.SubmitSyncCommitteeMessages(ctx, msgs)
+	if err != nil {
+		return err
+	}
+
+	log.Info(ctx, "Mock sync committee msg submitted", z.Int("slot", int(slot)), z.Str("root", hex.EncodeToString(blockRoot[:])))
```
Comment (corverroos) [inline testutil/validatormock/synccomm.go:235]: remove root, it isn't human understandable

File: testutil/beaconmock/headproducer.go [hunk 4]
```diff
@@ -0,0 +1,186 @@
...
+		w.WriteHeader(http.StatusInternalServerError)
+		_, _ = w.Write([]byte(`{"code": 500, "message": "Head producer not ready"}`))
+
+		return
+	}
+
+	blockID := mux.Vars(r)["block_id"]
+	if blockID != "head" && blockID != fmt.Sprint(head.Slot) {
+		w.WriteHeader(http.StatusBadRequest)
+		_, _ = w.Write([]byte(fmt.Sprintf(`{"code": 400, "message": "Invalid block ID: %s"}`, blockID)))
```
[the remark this answers is not in the document]
Comment (corverroos) [reply testutil/beaconmock/headproducer.go:128]: you'll need to address this lint

**Review by corverroos — APPROVED** [review]

**Pull request merged**